src/lib/y2network/interface_type.rb

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Network
+  # This class represents the interface types which are supported.
+  #
+  # Constants may be defined using the {define_type} method.
+  class InterfaceType
```

Comment:
good, I already have TODO for having such ENUM. So what is missing here? for sure old type string that is used in sysconfig like "eth", "wlan" and so on, so we can easily get from old type this new type. Also ideally if you can add all types now, so we can use it in other parts easily. Idea is to start using this new objects as soon as possible, so it won't be so separated.